db/migrate/20171026114327_add_deleted_on_to_container_quota_and_items.rb

```diff
@@ -1,9 +1,34 @@
 class AddDeletedOnToContainerQuotaAndItems < ActiveRecord::Migration[5.0]
+  class ContainerQuota < ActiveRecord::Base; end
+  class ContainerQuotaItem < ActiveRecord::Base
+    belongs_to :container_quota
+  end
+
   def change
     add_column :container_quotas, :deleted_on, :datetime
     add_index :container_quotas, :deleted_on
 
-    add_timestamps :container_quota_items
+    # Historical note: this migration has been edited. Original just naively added NOT
+    # NULL timestamp columns and failed with non-empty container_quota_items table.
```

Comment:
comment is not necessary, as the tests make it clear why this code is here.  Luckily, anybody with container_quota data can't get past this migration anyway, so we can't have any devs in an in-between state.